ci-operator/step-registry/assisted/baremetal/test/assisted-baremetal-test-commands.sh

```diff
@@ -26,32 +25,53 @@ trap collect_artifacts EXIT TERM
 # Tests execution
 set +e
 
-echo "### Copying test-list file"
-scp "${SSHOPTS[@]}" "${SHARED_DIR}/test-list" "root@${IP}:/tmp/test-list"
-
 echo "### Running tests"
-timeout --kill-after 10m 120m ssh "${SSHOPTS[@]}" "root@${IP}" bash -s "${OPENSHIFT_TESTS_IMAGE}" << "EOF"
+timeout --kill-after 10m 120m ssh "${SSHOPTS[@]}" "root@${IP}" "bash -s \"${OPENSHIFT_TESTS_IMAGE}\" \"${TEST_TYPE:-"list"}\" \"${TEST_SUITE:-"openshift/conformance/parallel"}\" \"${TEST_LIST:-""}\" \"${TEST_PROVIDER:-"baremetal"}\" \"${TEST_SKIPS}\"" << "EOF"
     set -x
 
-    function get_test_list() {
-        podman run --network host --rm -i -e KUBECONFIG=/tmp/kubeconfig -v ${kubeconfig}:/tmp/kubeconfig $1 \
-            openshift-tests run "openshift/conformance/parallel" --dry-run | \
-            grep -Ff /tmp/test-list
+    openshift_tests_image="$1"
+    test_type="$2"
+    test_suite="$3"
+    test_list="$4"
+    test_provider="$5"
+    test_skips="$6"
+
+    function get_baremetal_test_list() {
+        podman run --network host --rm -i \
+            -e KUBECONFIG=/tmp/kubeconfig -v "${KUBECONFIG}:/tmp/kubeconfig" "$1" \
+            openshift-tests run "$2" \
+            --dry-run \
+            --provider "{\"type\": \"$3\"}"
     }
 
     function run_tests() {
-        podman run --network host --rm -i -v /tmp:/tmp -e KUBECONFIG=/tmp/kubeconfig -v ${kubeconfig}:/tmp/kubeconfig $1 \
-            openshift-tests run -o /tmp/artifacts/e2e_${name}.log --junit-dir /tmp/artifacts/reports -f -
+        podman run --network host --rm -i -v /tmp:/tmp -e KUBECONFIG=/tmp/kubeconfig -v "${KUBECONFIG}:/tmp/kubeconfig" "$1" \
+            openshift-tests run -o "/tmp/artifacts/e2e_${name}.log" --junit-dir /tmp/artifacts/reports --file "$2"
     }
     
     # prepending each printed line with a timestamp
     exec > >(awk '{ print strftime("[%Y-%m-%d %H:%M:%S]"), $0 }') 2>&1
 
-    for kubeconfig in $(find ${KUBECONFIG} -type f); do
-        export KUBECONFIG=${kubeconfig}
-        name=$(basename ${kubeconfig})
-
-        stderr=$( { get_test_list $1 | run_tests $1; } 2>&1 )
+    for kubeconfig in $(find "${KUBECONFIG}" -type f); do
+        export KUBECONFIG="${kubeconfig}"
+        name=$(basename "${kubeconfig}")
+
+        case ${test_type} in
+            suite)
+                test_list=$(get_baremetal_test_list "${openshift_tests_image}" "${test_suite}" "${test_provider}")
```

Comment:
remove the params and use the named variable
```suggestion
                test_list=$(get_baremetal_test_list)
```